modules/objdetect/test/test_qrcode.cpp

```diff
@@ -458,56 +353,25 @@ TEST_P(Objdetect_QRCode_Curved, regression)
 #else
     ASSERT_TRUE(qrcode.detect(src, corners));
 #endif
-
-    const std::string dataset_config = findDataFile(root + "dataset_config.json");
-    FileStorage file_config(dataset_config, FileStorage::READ);
-    ASSERT_TRUE(file_config.isOpened()) << "Can't read validation data: " << dataset_config;
-    {
-        FileNode images_list = file_config["test_images"];
-        size_t images_count = static_cast<size_t>(images_list.size());
-        ASSERT_GT(images_count, 0u) << "Can't find validation data entries in 'test_images': " << dataset_config;
-
-        for (size_t index = 0; index < images_count; index++)
-        {
-            FileNode config = images_list[(int)index];
-            std::string name_test_image = config["image_name"];
-            if (name_test_image == name_current_image)
-            {
-                for (int i = 0; i < 4; i++)
-                {
-                    int x = config["x"][i];
-                    int y = config["y"][i];
-                    EXPECT_NEAR(x, corners[i].x, pixels_error);
-                    EXPECT_NEAR(y, corners[i].y, pixels_error);
-                }
-
-#ifdef HAVE_QUIRC
-                std::string original_info = config["info"];
-                EXPECT_EQ(decoded_info, original_info);
-#endif
-
-                return; // done
-            }
-        }
-        std::cerr
-            << "Not found results for '" << name_current_image
-            << "' image in config file:" << dataset_config << std::endl
-            << "Re-run tests with enabled UPDATE_QRCODE_TEST_DATA macro to update test data."
-            << std::endl;
-    }
+    check_qr(root, name_current_image, "test_images", corners, {decoded_info}, pixels_error);
 }
 
-typedef testing::TestWithParam < std::string > Objdetect_QRCode_Multi;
+typedef testing::TestWithParam<std::tuple<std::string, QRDetectMethod>> Objdetect_QRCode_Multi;
 TEST_P(Objdetect_QRCode_Multi, regression)
 {
-    const std::string name_current_image = GetParam();
+    const std::string name_current_image = get<0>(GetParam());
     const std::string root = "qrcode/multiple/";
+    const QRDetectMethod method = get<1>(GetParam());
     const int pixels_error = 4;
 
     std::string image_path = findDataFile(root + name_current_image);
     Mat src = imread(image_path);
     ASSERT_FALSE(src.empty()) << "Can't read image: " << image_path;
-    QRCodeDetector qrcode;
+    if (dropped_samples.find({name_current_image, method}) != dropped_samples.end())
+        return;
```

Comment:
Please throw `SkipTestException` exception. The test will be listed in reports and be marked as sipped by Google test.

Reply:
added `throw SkipTestException(name_current_image + " is disabled sample for method " + std::to_string((int)method));`